app/models/knight.py

```diff
@@ -0,0 +1,70 @@
+from typing import List, Optional
+
+
+class Armour:
+    def __init__(self, part: str, protection: int) -> None:
+        self.part = part
+        self.protection = protection
+
+
+class Weapon:
+    def __init__(self, name: str, power: int) -> None:
+        self.name = name
+        self.power = power
+
+
+class Potion:
+    def __init__(self, name: str, effect: dict) -> None:
+        self.name = name
+        self.effect = effect
+
+
+class Knight:
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        armour: Optional[List[Armour]] = None,
+        weapon: Optional[Weapon] = None,
+        potion: Optional[Potion] = None
+    ) -> None:
+        self.name = name
+        self.base_power = power
+        self.base_hp = hp
+        self.armour = armour or []
```

Comment:
The `armour` attribute is initialized with an empty list if not provided. This is correct, but ensure that the rest of the code handles an empty list appropriately, especially when calculating protection.